aws-examples/hackweek-infrastructure/icesat2-hackweek-2020-notes.md

```diff
@@ -0,0 +1,83 @@
+# Notes on JupyterHub infrastructure for the IceSat-2 2020 Hackweek
+
+Relevant links:
+- [JupyterHub Configuration](https://github.com/ICESAT-2HackWeek/jupyterhub-2020)
+- [Infrastructure Configuration](https://github.com/ICESAT-2HackWeek/terraform-deploy/tree/master)
+- [Computational Environment](https://github.com/ICESAT-2HackWeek/jupyter-image-2020 )
+
+Resource requests per-user:
+```
+    cpu:
+      limit: 4
+      guarantee: 1.75
+    memory:
+      limit: 8G
+      guarantee: 7G
+```
+
+Note m5 and m4 2xlarge nodes, so 4 people per node. 100Gb local disk
++ EFS drive with default settings. CPUs are logical cores, so the
+limit of 4 is 2 physical cpu cores with each with 2 independent
+threads.
+
+### Successes
+- Generally very stable for 80 simultaneous users! 
+- Scale-up time less then 5 minutes per person
+
+### Issues
+- Scale down an issue when people do not explicitly log out. Common
+to wake up to 5-10 nodes running for over 12 hours. 
+- Some complaints regarding performance when tutorials read the same
+data on `/srv/shared` network storage location
+- The `aws-node-termination-handler` will correctly replace user
+nodes if the nodes need replacing but users will not know why this is
+happening nor be able to do anything about it.
+
+### Lessons Learned / Future Hackweeks
+
+- Having prometheus / grafana is great to understand how CPU, RAM,
+network activity changes over time, and can be used to diagnose
+performance issues for individual users or nodes. Could be helpful to
+make these visualizations public and come up with custom dashboards 
+like instantaneous CPU % versus default averages over 10 minutes)
+  - Grafana probably needs a separate login per person who wants to
+  login simultaneously
+  - More practice with Grafana and more tailored dashboarding will
+  help admins to utilize Grafana usefully. I’ve spent a lot of time
```

Comment:
Would be great to capture any progress you made in working with Grafana if it would inform someone trying to do this in the future.